homeassistant/components/yi/camera.py

```diff
@@ -25,7 +25,7 @@
 
 DEFAULT_BRAND = "YI Home Camera"
 DEFAULT_PASSWORD = ""
-DEFAULT_PATH = "/tmp/sd/record"
+DEFAULT_PATH = "/tmp/sd/record"  # nosec
```

Comment:
Out of curiosity, what security issue is `bandit` seeing with this? A hardcoded filepath?

Reply:
https://bandit.readthedocs.io/en/latest/plugins/b108_hardcoded_tmp_directory.html

A hardcoded `/tmp` is a sign of possible incorrect temporary file handling. In most general cases such a path should not be hardcoded in the first place, but things like the `tempfile` module used instead so that environment settings such as `TMPDIR` are honored. If one uses a hardcoded tmp dir, it's not far fetched that one might also be overlooking other considerations related to secure temp file handling.

Reply:
Great info! Thanks. 👍🏻